hummingbot/client/performance_analysis.py

```diff
@@ -80,3 +100,133 @@ def compute_return(self, price: float) -> float:
+        elif trade.trade_type == TradeType.BUY.name:
+            net_base_delta: float = trade.amount * (1 - float(trade_fee["percent"])) - total_flat_fees
+            net_quote_delta: float = trade.amount * trade.price
+        else:
+            raise Exception(f"Unsupported trade type {trade.trade-type}")
+        return net_base_delta, net_quote_delta
+
+    def calculate_asset_delta_from_trades(self,
+                                          analysis_start_time: int,
+                                          current_startegy_name: str,
+                                          market_trading_pair_tuples: List[MarketTradingPairTuple]
+                                          ) -> Dict[MarketTradingPairTuple, Dict[str, float]]:
+        market_trading_pair_stats: Dict[MarketTradingPairTuple, Dict[str, float]] = {}
+        for market_trading_pair_tuple in market_trading_pair_tuples:
+            asset_stats: Dict[str, float] = {
+                market_trading_pair_tuple.base_asset.upper(): {"spent": s_float_0, "acquired": s_float_0},
+                market_trading_pair_tuple.quote_asset.upper(): {"spent": s_float_0, "acquired": s_float_0}
+            }
+            queried_trades: List[TradeFill] = TradeFill.get_trades(self.sql_manager.get_shared_session(),
+                                                                   start_time=analysis_start_time,
+                                                                   market=market_trading_pair_tuple.market.display_name,
+                                                                   strategy=current_startegy_name)
+            if not queried_trades:
+                market_trading_pair_stats[market_trading_pair_tuple] = {
+                    "starting_quote_rate": market_trading_pair_tuple.get_mid_price(),
+                    "asset": asset_stats
+                }
+                continue
+
+            for trade in queried_trades:
+                trade_side: str = trade.trade_type
+                base_asset: str = trade.base_asset.upper()
+                quote_asset: str = trade.quote_asset.upper()
+                base_delta, quote_delta = PerformanceAnalysis.calculate_trade_asset_delta_with_fees(trade)
+                if trade_side == TradeType.SELL.name:
+                    asset_stats[base_asset]["spent"] += base_delta
+                    asset_stats[quote_asset]["acquired"] += quote_delta
+                elif trade_side == TradeType.BUY.name:
+                    asset_stats[base_asset]["acquired"] += base_delta
+                    asset_stats[quote_asset]["spent"] += quote_delta
+
+            market_trading_pair_stats[market_trading_pair_tuple] = {
+                "starting_quote_rate": queried_trades[0].price,
+                "asset": asset_stats
+            }
+
+        return market_trading_pair_stats
+
+    def calculate_trade_performance(self,
```

Comment:
For larger functions like this, could you add some in-code comments?
(I know we don't really have a standard now, but would be good to start doing that in case someone needs to make changes in the future).